ThinDownloadManager/src/main/java/com/thin/downloadmanager/DownloadDispatcher.java

```diff
@@ -30,40 +30,53 @@
 import static java.net.HttpURLConnection.HTTP_SEE_OTHER;
 import static java.net.HttpURLConnection.HTTP_UNAVAILABLE;
 
+/**
+ * This thread/class used to make {@link HttpURLConnection}, receives Response from server
+ * and Dispatch the response to respective {@link DownloadRequest}
+ *
+ * @author Mani Selvaraj
+ * @author Praveen Kumar
+ */
 class DownloadDispatcher extends Thread {
 
-    /** The queue of download requests to service. */
+    /**
+     * The queue of download requests to service.
+     */
     private final BlockingQueue<DownloadRequest> mQueue;
-
-    /** Used to tell the dispatcher to die. */
-    private volatile boolean mQuit = false;
-
-
-
-    /** To Delivery call back response on main thread */
-    private DownloadRequestQueue.CallBackDelivery mDelivery;
-
-    /** The buffer size used to stream the data */
+    /**
+     * The buffer size used to stream the data
+     */
     private final int BUFFER_SIZE = 4096;
-
-    /** How many times redirects happened during a download request. */
-    private int mRedirectionCount = 0;
-
-    /** The maximum number of redirects. */
+    /**
+     * The maximum number of redirects.
+     */
     private final int MAX_REDIRECTS = 5; // can't be more than 7.
-
     private final int HTTP_REQUESTED_RANGE_NOT_SATISFIABLE = 416;
     private final int HTTP_TEMP_REDIRECT = 307;
-
+    /**
+     * Used to tell the dispatcher to die.
+     */
+    private volatile boolean mQuit = false;
+    /**
+     * To Delivery call back response on main thread
+     */
+    private DownloadRequestQueue.CallBackDelivery mDelivery;
+    /**
+     * How many times redirects happened during a download request.
+     */
+    private int mRedirectionCount = 0;
     private long mContentLength;
-    private long mCurrentBytes;
     private boolean shouldAllowRedirects = true;
 
+    private long mDownloadedCacheSize = 0;
```

Comment:
A comment for `mDownloadedCacheSize`

Reply:
A comment has been added to this variable, That will describe shortly about this variable that will tell the reason of variable why it appears here.